### DISCUSSION TITLE
Request: Native HomeAssisstant Add-On / Integration support

### DISCUSSION THREAD

**Discussion by millercentral** [teslamate-org/teslamate#467] [Ideas]
While I love the fact that it is possible to integrate TeslaMate data with HomeAssistant using MQTT data overrides to the existing Tesla integration, it feels like a hack, and is pretty difficult for a typical HomeAssistant RPi user to setup.

It would be amazing if you could create a variant of your docker that is a proper Home Assistant AddOn package. The documentation for their package system (which is docker under the covers) is available here: https://developers.home-assistant.io/docs/en/hassio_addon_index.html 

This package would likely need to combine the teslamate and postgres containers, but ideal would communicate directly with the HA Websocket API instead of MQTT. Additionally, once the addon is added to a HA system, it would have the ability to create a "discovered" Integration component for the car to get the entities setup. The Mosquito broker HA Add-on is a good example of how this works.

Again, I know HA support is not blocked without this, but this was greatly increase the usability of your awesome work for HA users like myself. Thanks for considering.

**Comment by matt-FFFFFF** [comment]
Hi. I have started looking at this.

I have created a separate postgres addon, which is working well. I am now developing the TeslaMate addon.

I'm having some issues with ingress, as TeslaMate doesn't seem to have the capability to run on a custom path. E.g. /foo/bar/

However I will drop ingress support in favour of the simpler and less secure port mapping. I will recommend that the external port is only enabled for setup purposes and then removed.

There is also a preexisting Grafana addon that I'm going to try. It will mean importing the dashboards but I think it's better than running two instances of Grafana.

**Comment by matt-FFFFFF** [comment]
I have this working and have created my own addon repo.

There are still some manual tasks, like uploading the Grafana dashboards, but it works:

https://github.com/matt-FFFFFF/hassio-addon-repository

**Comment by matt-FFFFFF** [comment]
Tagging #494 as a dependency

**Comment by DariBer** [comment]
I have installed the TeslaMate and PostgreSQL add-on. But I have problem with the db-connection, what should I write in the config for TeslaMate
database_host: abcdefgh-postgres
where do I find the hostname, I have tried with the ip och my Home Assistant, I have tried with localhost but doesn't work. So how can I find out what the hostname of my PostgreSQL add-on is?

**Comment by matt-FFFFFF** [comment]
Hi,

Apologies. The slug name is dependent on the repo that it comes from. When writing the docs I didn't know this, but the correct name is:

29b65938-postgres

________________________________

**Comment by matt-FFFFFF** [comment]
If you have an issue with my Postgres or Teslamate Home Assistant addons, please raise it here:

https://github.com/matt-FFFFFF/hassio-addon-repository

@adriankumpf does not support them

**Comment by matt-FFFFFF** [comment]
Thanks to @adriankumpf for the info supplied, I have now updated my addon now and it should now support ARMv7 and ARM64/aarch64. e.g. Raspberry Pi. I haven't got anything to test this on so please raise an issue on my repo if it doesn't work.

I am now using docker buildx and the original images produced here as a base, only adding a couple of things to support Home Assistant.

**Comment by quach128** [comment]
Hi Matt,

i added this to my hassio.  On starting the addon, im getting these errors ..

`[13:54:28] INFO: Checking for Grafana folder: TeslaMate
[13:54:28] INFO: ... RESTORED visited.json
[13:54:28] INFO: ... RESTORED charges.json
[13:54:28] INFO: ... RESTORED trip.json
[13:54:28] INFO: ... RESTORED drive-stats.json
[13:54:28] INFO: ... RESTORED charging-stats.json
[13:54:28] INFO: ... RESTORED projected-range.json
[13:54:28] INFO: ... RESTORED updates.json
[13:54:28] INFO: ... RESTORED statistics.json
[13:54:28] INFO: ... RESTORED locations.json
[13:54:28] INFO: ... RESTORED efficiency.json
[13:54:28] INFO: ... RESTORED charge-level.json
[13:54:28] INFO: ... RESTORED mileage.json
[13:54:28] INFO: ... RESTORED vampire-drain.json
[13:54:28] INFO: ... RESTORED overview.json
[13:54:28] INFO: ... RESTORED drives.json
[13:54:28] INFO: ... RESTORED states.json
[13:54:28] INFO: ... RESTORED drive-details.json
[13:54:29] INFO: ... RESTORED charge-details.json
[13:54:29] INFO: Finished Importing Grafana Dashboards
13:54:32.610 [info] Migrations already up
13:54:42.638 [info] Version: 1.20.1
13:54:42.664 [info] Running TeslaMateWeb.Endpoint with cowboy 2.8.0 at :::4000 (http)
13:54:42.665 [info] Access TeslaMateWeb.Endpoint at http://localhost
13:54:43.332 [error] GenServer {Tortoise.Registry, {Tortoise.Connection, "TESLAMATE_302E3336313"}} terminating
** (stop) {:connection_failed, :not_authorized}
Last message: :connect
13:54:43.334 [error] GenServer {Tortoise.Registry, {Tortoise.Connection, "TESLAMATE_302E3336313"}} terminating
** (stop) {:connection_failed, :not_authorized}
Last message: :connect
13:54:43.335 [error] GenServer {Tortoise.Registry, {Tortoise.Connection, "TESLAMATE_302E3336313"}} terminating
** (stop) {:connection_failed, :not_authorized}
Last message: :connect
13:54:43.336 [error] GenServer {Tortoise.Registry, {Tortoise.Connection, "TESLAMATE_302E3336313"}} terminating
** (stop) {:connection_failed, :not_authorized}
Last message: :connect
13:54:43.338 [error] GenServer {Tortoise.Registry, {Tortoise.Connection, "TESLAMATE_302E3335373"}} terminating
** (stop) {:connection_failed, :not_authorized}
Last message: :connect
13:54:43.339 [error] GenServer {Tortoise.Registry, {Tortoise.Connection, "TESLAMATE_302E3335373"}} terminating
** (stop) {:connection_failed, :not_authorized}
Last message: :connect
13:54:43.342 [error] GenServer {Tortoise.Registry, {Tortoise.Connection, "TESLAMATE_302E3335373"}} terminating
** (stop) {:connection_failed, :not_authorized}
Last message: :connect
13:54:43.344 [error] GenServer {Tortoise.Registry, {Tortoise.Connection, "TESLAMATE_302E3335373"}} terminating
** (stop) {:connection_failed, :not_authorized}
Last message: :connect
13:54:43.346 [error] GenServer {Tortoise.Registry, {Tortoise.Connection, "TESLAMATE_302E3731343"}} terminating
** (stop) {:connection_failed, :not_authorized}
Last message: :connect
13:54:43.348 [error] GenServer {Tortoise.Registry, {Tortoise.Connection, "TESLAMATE_302E3731343"}} terminating
** (stop) {:connection_failed, :not_authorized}
Last message: :connect
13:54:43.349 [error] GenServer {Tortoise.Registry, {Tortoise.Connection, "TESLAMATE_302E3731343"}} terminating
** (stop) {:connection_failed, :not_authorized}
Last message: :connect
13:54:43.351 [error] GenServer {Tortoise.Registry, {Tortoise.Connection, "TESLAMATE_302E3731343"}} terminating
** (stop) {:connection_failed, :not_authorized}
Last message: :connect
13:54:43.353 [error] GenServer {Tortoise.Registry, {Tortoise.Connection, "TESLAMATE_302E3739363"}} terminating
** (stop) {:connection_failed, :not_authorized}
Last message: :connect
13:54:43.354 [error] GenServer {Tortoise.Registry, {Tortoise.Connection, "TESLAMATE_302E3739363"}} terminating
** (stop) {:connection_failed, :not_authorized}
Last message: :connect
13:54:43.356 [error] GenServer {Tortoise.Registry, {Tortoise.Connection, "TESLAMATE_302E3739363"}} terminating
** (stop) {:connection_failed, :not_authorized}
Last message: :connect
13:54:43.357 [error] GenServer {Tortoise.Registry, {Tortoise.Connection, "TESLAMATE_302E3739363"}} terminating
** (stop) {:connection_failed, :not_authorized}
Last message: :connect
13:54:43.361 [info] Application teslamate exited: shutdown
Kernel pid terminated (application_controller) ({application_terminated,teslamate,shutdown})
{"Kernel pid terminated",application_controller,"{application_terminated,teslamate,shutdown}"}

Crash dump is being written to: erl_crash.dump...done`

**Comment by matt-FFFFFF** [comment]
Hi,

Tortoise is the MQTT client. Please check the details or set mqtt to be false and try again. Raise an issue on my repo if this persists.

Matt
________________________________
